polygon_inference.py

```diff
@@ -544,10 +599,10 @@ def point_in_polygon(point, polygon):
                 return False
             # Convert polygon to the format expected by cv2.pointPolygonTest
             poly_points = polygon.astype(np.float32).reshape((-1, 1, 2))
-            return cv2.pointPolygonTest(poly_points, point, False) >= 0
+            return cv2.pointPolygonTest(poly_points, point, True) >= -2
```

Comment:
This line is the most important change: 2 pixels of tolerance for matching buildings across tiles.